docs/_integration-with-existing-apps-ios.md

```diff
@@ -0,0 +1,598 @@
++    [button.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
++    [button.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
++    [button.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor],
++    [button.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
++  ]];
+}
+
++- (void)presentReactNative
++{
++  [self presentViewController:[ReactViewController new] animated:YES completion:nil];
++}
+
+@end
+```
+
+</TabItem>
+<TabItem value="swift">
+
+```diff title="ViewController.swift"
+import UIKit
+
+class ViewController: UIViewController {
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    // Do any additional setup after loading the view.
+    self.view.backgroundColor = .systemBackground
+
++    let button = UIButton()
++    button.setTitle("Open React Native", for: .normal)
++    button.setTitleColor(.systemBlue, for: .normal)
++    button.setTitleColor(.blue, for: .highlighted)
++    button.addAction(UIAction { [weak self] _ in
++      self?.present(ReactViewController(), animated: true)
++    }, for: .touchUpInside)
++    self.view.addSubview(button)
++
++    button.translatesAutoresizingMaskIntoConstraints = false
++    NSLayoutConstraint.activate([
++      button.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
++      button.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
++      button.centerXAnchor.constraint(equalTo: self.view.centerXAnchor),
++      button.centerYAnchor.constraint(equalTo: self.view.centerYAnchor),
++    ])
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+3. Update the `ReactViewController` code as it follows:
+
+<Tabs groupId="ios-language" queryString defaultValue={constants.defaultAppleLanguage} values={constants.appleLanguages}>
+<TabItem value="objc">
+
+```diff title="ReactViewController.m"
+#import "ReactViewController.h"
++#import <React-RCTAppDelegate/RCTRootViewFactory.h>
++#import <React-RCTAppDelegate/RCTAppDelegate.h>
+
+@interface ReactViewController ()
+
+@end
+
+@implementation ReactViewController
+
+ - (void)viewDidLoad {
+   [super viewDidLoad];
+   // Do any additional setup after loading the view.
++   RCTRootViewFactory * factory = ((RCTAppDelegate *)RCTSharedApplication().delegate).rootViewFactory;
++   self.view = [factory viewWithModuleName:@"HelloWorld"];
+ }
+
+@end
+```
+
+</TabItem>
+<TabItem value="swift">
+
+```diff title="ReactViewController.swift"
+import UIKit
++import React_RCTAppDelegate
+
+class ReactViewController: UIViewController {
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
++    let factory = (RCTSharedApplication()?.delegate as? RCTAppDelegate)?.rootViewFactory;
++    self.view = factory?.view(withModuleName: "HelloWorld");
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+4. Make sure to disable the Sandbox scripting. To achieve this, in Xcode, click on your app, then on build settings. Filter for script and set the `User Script Sandboxing` to `NO`.
```

Comment:
Can you add a sentence about why you disable Sandbox scripting?  If it isn't important please ignore, just the first question that popped into my head.

Reply:
Sure, it's for Hermes, anyway.